core/logging.py

```diff
@@ -52,27 +46,95 @@ def emit(self, record: logging.LogRecord):
         )
 
 
-oldLevels = {k: v.getEffectiveLevel()
-             for k, v in logging.root.manager.loggerDict.items()
-             if isinstance(v, logging.Logger)}
+# ViURLocalFormatter ---------------------------------------------------------------------------------------------------
+
+class ViURLocalFormatter(logging.Formatter):
+    """
+    This is a formatter that injects console color sequences for debug output.
+    """
+
+    BLACK, RED, GREEN, YELLOW, BLUE, MAGENTA, CYAN, WHITE = range(8)
+    RESET_SEQ = "\033[0m"
+    COLOR_SEQ = "\033[1;%dm"
+    BOLD_SEQ = "\033[1m"
+
+    LEVELS = {
+        "WARNING": YELLOW,
+        "INFO": CYAN,
+        "DEBUG": BLUE,
+        "CRITICAL": RED,
+        "ERROR": RED
+    }
+
+    def format(self, record):
+        if "pathname" in record.__dict__.keys():
+            # truncate the pathname
+            if "/deploy" in record.pathname:
+                pathname = record.pathname.split("/deploy/")[1]
+            else:
+                pathname = record.pathname
+                if len(pathname) > 20:
+                    pathname = ".../" + "/".join(pathname.split("/")[-3:])
+
+            record.pathname = pathname
+
+        levelname = record.levelname
+
+        if levelname in ViURLocalFormatter.LEVELS:
+            record.levelname = ViURLocalFormatter.COLOR_SEQ % (30 + ViURLocalFormatter.LEVELS[levelname]) \
+                               + levelname + ViURLocalFormatter.RESET_SEQ
```

Comment:
I like it when the whole line is colored instead of just parts of it. While reading the message (on the right), you notice the color and know what kind of logging it is without having to read the logging level (further to the left) itself.
Also, this has the nice effect of adding a kind of even/odd style (often used in tables) to separate rows and improve readability (of course only works if the level are different).

Reply:
I've fixed this comment to colorize the entire line.

Reply:
Hi I just stumbled across this PR. Is it possible to have it configurable if the whole line or only the level should be colored? I'm thinking of terminals with white background (mac default). I think if the whole line is colored it will be hard to read.